pkg/pipeline/extract/conntrack/conn.go

```diff
@@ -106,6 +109,64 @@ func (c *connType) markReported() bool {
 	return isFirst
 }
 
+func (c *connType) isMatchSelectorStrings(selector map[string]string) bool {
+	for k, v := range selector {
+		connValueRaw, found := c.keys[k]
+		if !found {
+			return false
+		}
+		if fmt.Sprintf("%v", connValueRaw) != v {
+			return false
+		}
+	}
+	return true
+}
+
+func (c *connType) isMatchSelectorGeneric(selector map[string]interface{}) bool {
+	for k, v := range selector {
+		connValueRaw, found := c.keys[k]
+		if !found {
+			return false
+		}
+		switch connValue := connValueRaw.(type) {
+		case int:
+			selectorValue, err := utils.ConvertToInt(v)
+			if err != nil || connValue != selectorValue {
+				return false
+			}
+		case uint32:
+			selectorValue, err := utils.ConvertToUint32(v)
+			if err != nil || connValue != selectorValue {
+				return false
+			}
+		case uint64:
+			selectorValue, err := utils.ConvertToUint64(v)
+			if err != nil || connValue != selectorValue {
+				return false
+			}
+		case int64:
+			selectorValue, err := utils.ConvertToInt64(v)
+			if err != nil || connValue != selectorValue {
+				return false
+			}
+		case float64:
+			selectorValue, err := utils.ConvertToFloat64(v)
+			if err != nil || connValue != selectorValue {
+				return false
+			}
+		case bool:
+			selectorValue, err := utils.ConvertToBool(v)
+			if err != nil || connValue != selectorValue {
+				return false
+			}
+		case string:
+		default:
+			return fmt.Sprintf("%v", c.keys[k]) != connValue
+		}
+	}
+	return true
```

Comment:
For the default case, if it comes out `true`, don't we have to continue the loop to check the remaining conditions?

Reply:
Good catch! My bad.
In addition, the `case string:` is also wrong, in Go there is no automatic fallthrough and in type assertion switch it's not allowed anyway. So, in case of strings, it simply ignores them. I'll add a fix and another test cases.